clients/src/main/java/org/apache/kafka/common/record/AbstractRecords.java

```diff
@@ -126,11 +172,43 @@ protected LogRecord makeNext() {
         };
     }
 
-    public static int estimatedSizeRecords(CompressionType compressionType, Iterable<LogRecord> records) {
+    public static int sizeEstimateInBytes(byte magic,
+                                          long baseOffset,
+                                          CompressionType compressionType,
+                                          Iterable<LogRecord> records) {
         int size = 0;
-        for (LogRecord record : records)
-            size += record.sizeInBytes();
-        // TODO: Account for header overhead
+        if (magic <= LogEntry.MAGIC_VALUE_V1) {
+            for (LogRecord record : records)
+                size += record.sizeInBytes();
```

Comment:
The redundancy is annoying, but the methods are different. A `LogRecord` has an offset which must be preserved. A `KafkaRecord` is an abstract record which doesn't exist in any log. It is possible to convert a `LogRecord` to a `KafkaRecord`, but the need to recopy the collection just to compute the size seems more annoying than the minimal duplication of code. One option I've considered is making `KafkaRecord` an interface and having `LogRecord` extend it. That's fine, but then I need to introduce _another_ record type to do what `KafkaRecord` does.